zcash_client_backend/src/data_api/testing/pool.rs

```diff
@@ -0,0 +1,90 @@
+use std::{cmp::Eq, hash::Hash};
+
+use incrementalmerkletree::Level;
+use rand::RngCore;
+use shardtree::error::ShardTreeError;
+use zcash_keys::{address::Address, keys::UnifiedSpendingKey};
+use zcash_primitives::transaction::Transaction;
+use zcash_protocol::{
+    consensus::{self, BlockHeight},
+    memo::MemoBytes,
+    value::Zatoshis,
+    ShieldedProtocol,
+};
+
+use crate::{
+    data_api::{
+        chain::{CommitmentTreeRoot, ScanSummary},
+        DecryptedTransaction, InputSource, WalletCommitmentTrees, WalletRead, WalletSummary,
+    },
+    wallet::{Note, ReceivedNote},
+};
+
+use super::{TestFvk, TestState};
+
+/// Trait that exposes the pool-specific types and operations necessary to run the
+/// single-shielded-pool tests on a given pool.
+pub trait ShieldedPoolTester {
```

Comment:
The trait's associated types and methods need documentation now that it is in the public API.

Reply:
Addressed in #1539.